### PR TITLE
Improve student activity registration system

### PR DESCRIPTION
**Pull request by am-kovalchuk**
This pull request introduces significant improvements to the FastAPI school activities app, focusing on participant management, frontend usability, and test coverage. The most important changes include adding the ability for users to unregister from activities, enhancing the frontend to display and manage participants interactively, expanding the set of activities, and introducing a comprehensive automated test suite to ensure reliability.

**Backend functionality enhancements:**

* Added an `unregister_from_activity` endpoint in `src/app.py` to allow users to remove themselves from activities, including validation to prevent duplicate removals and handle non-existent activities.
* Improved signup logic to prevent duplicate signups for the same activity.

**Frontend improvements:**

* Updated `src/static/app.js` to display a list of current participants for each activity, including a delete button for each participant to trigger unregistration. Added logic to refresh the activities list after signup/unregister actions. [[1]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R16-R47) [[2]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R85-R88) [[3]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R108-R152)
* Enhanced styles in `src/static/styles.css` for participant lists, items, and delete buttons, improving the overall UI experience.

**Activity data expansion:**

* Added several new activities (sports, artistic, intellectual) to the `activities` dictionary in `src/app.py`, increasing the diversity of available options for students.

**Testing and developer tooling:**

* Added a `run_tests.sh` script to simplify running tests with coverage reporting.
* Created `tests/conftest.py` for pytest fixtures to ensure test isolation and provide sample data.
* Developed a comprehensive test suite in `tests/test_api.py` covering all endpoints, edge cases, and data integrity scenarios.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+def client():
+    """Create a test client for the FastAPI application."""
+    return TestClient(app)
+
+
+@pytest.fixture(autouse=True)
+def reset_activities():
+    """Reset activities data before each test to ensure test isolation."""
+    # Store original activities with deep copy to preserve nested structures
+    original_activities = copy.deepcopy(activities)
```
[the remark this answers is not in the document]
Comment (am-kovalchuk) [reply tests/conftest.py]:
```suggestion
    activities.update({
```
